GoogleUtilities/AppDelegateSwizzler/GULAppDelegateSwizzler.m

```diff
@@ -1002,6 +1064,89 @@ + (void)proxyAppDelegate:(id<GULApplicationDelegate>)appDelegate {
   }
 }
 
+#if UISCENE_SUPPORTED
++ (void)proxySceneDelegate:(UIScene *)scene {
+  Class realClass = [scene.delegate class];
+
+  // Skip proxying if the class has a prefix of kGULAppDelegatePrefix, which means it has been
+  // proxied before.
+  if ([NSStringFromClass(realClass) hasPrefix:kGULAppDelegatePrefix]) {
+    return;
+  }
+
+  NSString *classNameWithPrefix =
+      [kGULAppDelegatePrefix stringByAppendingString:NSStringFromClass(realClass)];
+  NSString *newClassName =
+      [NSString stringWithFormat:@"%@-%@", classNameWithPrefix, [NSUUID UUID].UUIDString];
+
+  if (NSClassFromString(newClassName)) {
+    GULLogError(kGULLoggerSwizzler, NO,
+                [NSString stringWithFormat:@"I-SWZ%06ld",
+                                           (long)kGULSwizzlerMessageCodeAppDelegateSwizzling005],
```

Comment:
nit: Not sure if we should reuse the same message code. I think we should create separate message codes for this method. Also the message should be updated to replace "App Delegate" by "Scene Delegate"

Reply:
Yeah, dedicated message codes should be better. Can I add it directly? or is there any review process for GULLogger?

Reply:
Feel free to add them [here](https://github.com/firebase/firebase-ios-sdk/blob/080397209ec36f20d08f8fc931115bcd8cf82886/GoogleUtilities/Common/GULLoggerCodes.h#L19). A meaningful name is the recent recommendation (like `kGULSwizzlerMessageCodeAppDelegateSwizzlingInvalidAppDelegate` in the enum).